pkgs/by-name/dc/dcmtk/package.nix

```
@@ -14,13 +15,51 @@
 stdenv.mkDerivation rec {
   pname = "dcmtk";
   version = "3.6.9";
+
   src = fetchFromGitHub {
     owner = "DCMTK";
-    repo = pname;
-    rev = "DCMTK-${version}";
+    repo = "dcmtk";
+    tag = "DCMTK-${version}";
     hash = "sha256-mdI/YqM38WhnCbsylIlmqLLWC5/QR+a8Wn9CNcN7KXU=";
   };
 
+  # The following patches are taken from the Debian package
+  # See https://salsa.debian.org/med-team/dcmtk
+  patches = [
+    (fetchpatch {
+      url = "https://salsa.debian.org/med-team/dcmtk/-/raw/master/debian/patches/01_dcmtk_3.6.0-1.patch";
```

```suggestion
      url = "https://salsa.debian.org/med-team/dcmtk/-/raw/debian/3.6.9-4/debian/patches/01_dcmtk_3.6.0-1.patch";
```

Can we pull the patches from the last tag instead of master just to be sure that are not deleted/modified later on?

Also, the patches can be retrieved directly with `fetchurl` since they are static files.

---

Very good remark, thanks! Going to make the changes.

---

What's the difference between using `fetchpatch` vs `fetchurl` in this case?

---

`fetchpatch` does a bit more work because it tries to remove all the possibly variable parts (e.g. the Git version that might be present) which is not really useful here.

If you keep the `fetchurl`, you also need to adjust the FOD hashes:
```
┏━ 8 Errors: 
┃ error: hash mismatch in fixed-output derivation '/nix/store/z234zxjpwpfibz0qmzsmjn989wy6qfp3-remove_version.patch.drv':
┃          specified: sha256-EAgmrR7jN+krzA0kKVQZRefKO0hU3/x8v1sTSYbEc44=
┃             got:    sha256-jcV2xQzKdNiBgcaFtaxdJpJCCSVOqGIsi/A4iqVM8U8=
┃ error: hash mismatch in fixed-output derivation '/nix/store/5iai72zyzkpiv9j0pg9iy4lds5zpll3z-07_dont_export_all_executables.patch.drv':
┃          specified: sha256-qrjf/d5ILXCvJKYA+12w+rP6fcAjcEU1jlaq9TaNNLE=
┃             got:    sha256-5slFod+S7Yuj0u2CfTUw+MWZYuqQs4hgoGmh3KAUo+c=
┃ error: hash mismatch in fixed-output derivation '/nix/store/h3y8k2ddykypnzpvdd2vhva1qilf9z3f-01_dcmtk_3.6.0-1.patch.drv':
┃          specified: sha256-QHdm7HORy9xhVqCj6dH0mbcDZISpfyFth2H9bl3L7tI=
┃             got:    sha256-kDEZvPqcF8+PYID24srMoPSBPltmnGiJ67LHsLVcPYM=
┃ error: hash mismatch in fixed-output derivation '/nix/store/ya686xjv2cs75qjan50cj3da2dgpiy0y-0011-CVE-2025-25472.patch.drv':
┃          specified: sha256-PzrA1UVxwtK0AWCEhSfH3Z+/DCoEXW03OAqT4zqVTQM=
┃             got:    sha256-o3/PykJFbYlasAFgPNWp09hRuH183tQuvGuaOV4MOoo=
┃ error: hash mismatch in fixed-output derivation '/nix/store/mgcg3183z28jz2gx3738vf7mycs1g6h2-0010-CVE-2025-25474.patch.drv':
┃          specified: sha256-SUeUdpwhSxBYEoNpf5wHCo8i1ZRF+AxwUtM/6cMs41o=
┃             got:    sha256-aX8em1o88ND4srsYkG696elPsAIlvkRRZMT8wzD2GdQ=
┃ error: hash mismatch in fixed-output derivation '/nix/store/7fcxk8iikaagsq14sj6xqxk5nr9k5kip-0009-CVE-2025-25475.patch.drv':
┃          specified: sha256-wmvpqteJCOQMkbMdRDclEEyGVFcr0dBEIe+ZHrNRESg=
┃             got:    sha256-ApuVw6aBoasuVlJ3fh/aufB2WRm2hFgLYCq1k3MPrsU=
┃ error: hash mismatch in fixed-output derivation '/nix/store/3s5xlgfbc2jkrhkgly9n24zjldqr3mvi-0008-CVE-2024-52333.patch.drv':
┃          specified: sha256-e42IU6ZjjA9a8uW3KSKz8hJa+KeGKXg4yodq3vljStM=
┃             got:    sha256-/4NdauuH0v6CPMh+duMM91wWfylp6l4L2LTO80dDh9g=
┃ error: hash mismatch in fixed-output derivation '/nix/store/61rsa8cpkji2a8vy1apscavz3v3i7a32-0007-CVE-2024-47796.patch.drv':
┃          specified: sha256-lXadDWpwDDT19OuBMwSC9KMNOV6o/Gx39XodG8v6h4s=
┃             got:    sha256-QYWgSbyIcOq3CVg2ynVSPCHBIrDj9uqX4ese1huoOoU=

```

---

Thanks, fixed!